### PR TITLE
Add windows Grafana agent installation script

### PR DESCRIPTION
**Pull request by mshahzeb**
Hi team along with the effort of upgrading the Windows integration - I have written a powershell script to automate the installation and configuration of Grafana agent on Windows. Similar to the linux agent script - the powershell script:

1. Downloads the Grafana agent windows installer
2. Installs the Grafana agent using the installer in silent mode
3. Gets the agent config from integration API and converts it to YAML
4. Copies over the config and restarts the agent

The steps needed are as below:
```
# Download the Powershell script - run the following command in command prompt:
curl -L https://storage.googleapis.com/grafanalabs-integration-assets/windows/scripts/grafanacloud-install.ps1 --output grafanacloud-install.ps1

# Then execute the script from a command prompt running with administrator privileges:
powershell -executionpolicy Bypass -File “.\grafanacloud-install.ps1" -GCLOUD_STACK_ID <Gcloud Stack ID> -GCLOUD_API_KEY <Gcloud API Key> -GCLOUD_API_URL <Gcloud API Url>
```

The script is written to be compatible with Powershell 3.0 and above

### CODE REVIEW

File: production/grafanacloud-install.ps1 [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+$AUTH_TOKEN = "Bearer $GCLOUD_API_KEY"
+
+$headers = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
+$headers.Add("Authorization", $AUTH_TOKEN)
+
+$response = Invoke-WebRequest $CONFIG_URI -Method 'GET' -Headers $headers
+
+$jsonObj = $response | ConvertFrom-Json
+if ($jsonObj.status -eq "success") {
+	Write-Host "Saving and updatig agent configuration file"
```
Comment (mattdurham) [inline production/grafanacloud-install.ps1:55]: nite: updating

**Review by mattdurham — CHANGES_REQUESTED** [review]
LGTM, may need to trigger a new build though to get the CI to pass

File: production/grafanacloud-install.ps1 [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+# Install Grafana agent in silent mode
+Write-Host "Installing Grafana agent for Windows"
+.\grafana-agent-installer.exe /S
+
+Write-Host "Retrieving and updating Grafana agent config"
+$CONFIG_URI = "$GCLOUD_API_URL/stacks/$GCLOUD_STACK_ID/agent_config"
+$AUTH_TOKEN = "Bearer $GCLOUD_API_KEY"
+
+$headers = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
```
[the remark this answers is not in the document]
Comment (mshahzeb) [reply production/grafanacloud-install.ps1]: Thanks! Updated :)

File: production/grafanacloud-install.ps1 [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+$headers.Add("Authorization", $AUTH_TOKEN)
+
+$response = Invoke-WebRequest $CONFIG_URI -Method 'GET' -Headers $headers
```
[the remark this answers is not in the document]
Comment (mshahzeb) [reply production/grafanacloud-install.ps1:52]: I had issues while using convert to json with Invoke-RestMethod - it has some auto conversion to json mechanism that was not working well, so reverted to usig Invoke-WebRequest.

**Review by mattdurham — APPROVED** [review]
Looks great

**Pull request merged**